### DISCUSSION TITLE
Is it bad practice to use browser, node fetch api or useSWR library in Remix?

### DISCUSSION THREAD

**Discussion by sanghogu** [remix-run/remix#8656] [Q&A]
I often use fetch or useSWR from time to time and would like to know whether this is a recommended action in Remix or not.

**Comment by kiliman** [comment]
Nothing prevents you from using those with Remix apps. In fact, you'll still need those if you're calling non-Remix routes from the client.

Ideally, it would be best to migrate to `loaders` and `useLoaderData` or `useFetcher`. You can also centralize your data loading using the new Client Data support. 

https://remix.run/docs/en/main/guides/client-data

**Comment by sanghogu** [comment]
thanks have a good day
